drivers/charger/charger_bq25180.c

```diff
@@ -87,6 +93,27 @@ static uint32_t bq25180_ichg_to_ma(uint8_t ichg)
 	return (ichg - 31) * 10 + 40;
 }
 
+static int bq25180_mv_to_vbatreg(uint32_t voltage_mv, uint8_t *vbat)
+{
+	if (!IN_RANGE(voltage_mv, BQ25180_VOLTAGE_MIN_MV, BQ25180_VOLTAGE_MAX_MV)) {
+		LOG_WRN("charging voltage out of range: %dmV, "
+			"clamping to the nearest limit",
+			voltage_mv);
+	}
+	voltage_mv = CLAMP(voltage_mv, BQ25180_VOLTAGE_MIN_MV, BQ25180_VOLTAGE_MAX_MV);
```

Comment:
Minor nit, it would be nicer to clamp the maximum to what is provided by `constant-charge-voltage-max-microvolt`. This would ensure the regulation voltage target never exceeds what is set higher than the maximum set in DT.

Reply:
Also renamed the variable from `initial_voltage_microvolt` to `max_voltage_microvolt`